server/schedule/placement/rule_list.go

```diff
@@ -98,15 +98,16 @@ type ruleList struct {
 	ranges []rangeRules // ranges[i] contains rules apply to [ranges[i].startKey, ranges[i+1].startKey).
 }
 
+type ruleContainer interface {
+	iterateRules(func(*Rule))
+}
```

Comment:
what about `rulesIterator`.

Reply:
em, I think `rulesIterator` usually has a `Next()` functions and returns an element every time `Next` is called. But here it works differently.

Reply:
got it.